src/java/org/apache/cassandra/net/GlobalBufferPoolAllocator.java

```diff
@@ -36,6 +36,6 @@ private GlobalBufferPoolAllocator()
 
     static ByteBuf wrap(ByteBuffer buffer)
```

Comment:
the use cases of this method are fixed-length buffer. should be fine to add `maxCapacity`

Reply:
@jasonstack How about using `buffer.capacity()` here then?
Since `maxCapacity` is not ignored anymore, I had to come up with a default value for this method, so I went with netty's default (Integer.MAX_VALUE) to be on the safe side, but if the use case here is fixed-length buffers and this is not an overridden method (it's only used internally), I guess we can go with `buffer.capacity()`.
If you think that's reasonable, I can push that change and run another build in CircleCI.

The main issue was in `BufferPoolAllocator#newDirectBuffer` which was called by netty but the maxCapacity param was ignored.

Reply:
work for me. thanks

Reply:
@jasonstack Changed pushed. Running new CI builds.